maven-resolver-provider/src/main/java/org/apache/maven/repository/internal/type/DefaultType.java

```diff
@@ -95,8 +101,8 @@ public Map<String, String> getProperties() {
         Map<String, String> properties = new HashMap<>();
         properties.put(ArtifactProperties.TYPE, this.id);
         properties.put(ArtifactProperties.LANGUAGE, this.language.id());
-        properties.put(ArtifactProperties.INCLUDES_DEPENDENCIES, String.valueOf(includesDependencies));
-        properties.put(ArtifactProperties.CONSTITUTES_BUILD_PATH, String.valueOf(buildPathConstituent));
+        properties.put(MavenArtifactProperties.INCLUDES_DEPENDENCIES, String.valueOf(includesDependencies));
+        properties.put(MavenArtifactProperties.CONSTITUTES_BUILD_PATH, String.valueOf(buildPathConstituent));
         return Collections.unmodifiableMap(properties);
```

Comment:
This map is immutable and could be computed in the `DefaultType` constructor

Reply:
fixed